framework/sandstone_opts.cpp

```diff
@@ -0,0 +1,911 @@
+                break;
+            case is_asan_option:
+            case is_debug_option:
+                // these options are only accessible in the command-line if the
+                // corresponding functionality is active
+                opts.action = Action::exit;
+                return EXIT_SUCCESS;
+            case max_cores_per_slice_option:
+                opts.max_cores_per_slice_ = ParseIntArgument<>{
+                        .name = "--max-cores-per-slice",
+                        .min = -1,
+                    }();
+                break;
+            case mce_check_period_option:
+                app->mce_check_period = ParseIntArgument<>{"--mce-check-every"}();
+                break;
+            case no_slicing_option:
+                opts.max_cores_per_slice_ = -1;
+                break;
+            case on_crash_option:
+                opts.on_crash_arg = optarg;
+                break;
+            case on_hang_option:
+                opts.on_hang_arg = optarg;
+                break;
+            case output_format_option:
+                if (strcmp(optarg, "key-value") == 0) {
+                    app->shmem->output_format = SandstoneApplication::OutputFormat::key_value;
+                } else if (strcmp(optarg, "tap") == 0) {
+                    app->shmem->output_format = SandstoneApplication::OutputFormat::tap;
+                } else if (strcmp(optarg, "yaml") == 0) {
+                    app->shmem->output_format = SandstoneApplication::OutputFormat::yaml;
+                } else if (SandstoneConfig::Debug && strcmp(optarg, "none") == 0) {
+                    // for testing only
+                    app->shmem->output_format = SandstoneApplication::OutputFormat::no_output;
+                    app->shmem->verbosity = -1;
+                } else {
+                    fprintf(stderr, "%s: unknown output format: %s\n", argv[0], optarg);
+                    return EX_USAGE;
+                }
+                break;
+
+            case quality_option:
+                app->requested_quality = ParseIntArgument<>{
+                        .name = "--quality",
+                        .min = int(TEST_QUALITY_SKIP),
+                        .max = int(TEST_QUALITY_PROD),
+                        .range_mode = OutOfRangeMode::Saturate
+                }();
+                break;
+
+            case quick_run_option:
+                app->max_test_loop_count = 1;
+                app->delay_between_tests = 0ms;
+                break;
+            case retest_on_failure_option:
+                app->retest_count = ParseIntArgument<>{
+                        .name = "--retest-on-failure",
+                        .max = SandstoneApplication::MaxRetestCount,
+                        .range_mode = OutOfRangeMode::Saturate
+                }();
+                break;
+            case reschedule_option:
+                if (app->thread_count < 1) {
```

Comment:
This is a bug, actually, it needs to be `< 2`. I know it wasn't introduced by this change, but please fix it.